iroh-dns-server/src/store.rs

```diff
@@ -55,12 +58,26 @@ impl ZoneStore {
         Ok(Self::new(packet_store))
     }
 
-    /// Configure a pkarr client for resolution of packets from the bittorrent
-    /// mainline DHT.
+    /// Configure a pkarr client for resolution of packets from the bittorent mainline DHT.
     ///
     /// This will be used only as a fallback if there is no local info available.
-    pub fn with_pkarr(self, pkarr: Option<Arc<PkarrClient>>) -> Self {
-        Self { pkarr, ..self }
+    ///
+    /// Optionally set custom bootstrap nodes. If `bootstrap` is empty it will use the default
+    /// mainline bootstrap nodes.
+    pub fn with_mainline_fallback(self, bootstrap: &[SocketAddr]) -> Self {
+        let pkarr_client = if bootstrap.is_empty() {
+            PkarrClient::default()
+        } else {
+            let bootstrap = bootstrap
+                .iter()
+                .map(|addr| addr.to_string())
```

Comment:
Can these be URLs, or why does pkarr require strings here? Hopefully that can be changed in pkarr 2.0

Reply:
Good point. I misread the pkarr source - these can be either "ip:port" or "domain:port" strings. changed to use strings on our end as well.